packages/gatsby/src/schema/infer/example-value.js

```diff
@@ -0,0 +1,165 @@
+const _ = require(`lodash`)
+const { reportConflict } = require(`./type-conflict-reporter`)
+const is32BitInteger = require(`./is-32-bit-integer`)
+const { isDate } = require(`../types/Date`)
+
+const getExampleValue = ({ nodes, typeName, ignoreFields }) => {
+  const exampleValue = getExampleObject({ nodes, typeName, ignoreFields })
+  return exampleValue
+}
+
+module.exports = {
+  getExampleValue,
+}
+
+const getExampleObject = ({ nodes, prefix, ignoreFields = [] }) => {
+  const allKeys = nodes.reduce(
+    (acc, node) =>
+      Object.keys(node).forEach(
+        key => key && !ignoreFields.includes(key) && acc.add(key)
+      ) || acc,
+    new Set()
+  )
+
+  const exampleValue = Array.from(allKeys).reduce((acc, key) => {
+    const entries = nodes
+      .map(node => {
+        const value = node[key]
+        const type = getType(value)
+        return type && { value, type, parent: node }
+      })
+      .filter(Boolean)
+
+    const selector = prefix ? `${prefix}.${key}` : key
+
+    const entriesByType = _.uniqBy(entries, entry => entry.type)
+    if (!entriesByType.length) return acc
+
+    // TODO: This whole thing could be prettier!
+
+    let { value, type } = entriesByType[0]
+    let arrayWrappers = 0
+    while (Array.isArray(value)) {
+      value = value[0]
+      arrayWrappers++
+    }
+
+    if (entriesByType.length > 1 || type.includes(`,`)) {
+      if (
+        isMixOfDatesAndStrings(
+          entriesByType.map(entry => entry.type),
+          arrayWrappers
+        )
+      ) {
+        // FIXME: is a mix of date *objects* and strings problematic?
+        // I.e. when date objects will be treated as strings,
+        // are they automatically stringified? Probably yes, because
+        // `Date` has `toString` method. In current master, we return
+        // InvalidValue.
+        value = `String`
```

Comment:
This is actually a change in behavior: we currently infer a mix of Date objects and strings as INVALID_VALUE

Reply:
I don't see a point of treating mix of dates/strings differently than other conflicting values and we don't have concept of INVALID_VALUE anymore, so I'll move it to normal conflicts.

Reply:
@pieh could you comment if this would be a correct behaviour?

Reply:
OK, we could just remove it and it passes tests, so I assume this behaviour works correctly.